File: examples/pingpong_demo/pong/pong.go
package main

import (
	"encoding/json"
	"fmt"
	"io"
	"log"

	"context"

	dappclient "inspr.dev/inspr/pkg/client"
	"inspr.dev/inspr/pkg/sidecars/models"
)

func main() {

	client := dappclient.NewAppClient()
	ctx, cancel := context.WithCancel(context.Background())
	defer cancel()

	sentMsg := "Pong!"
	if err := client.WriteMessage(ctx, "pongoutput", sentMsg); err != nil {
		fmt.Printf("an error occurred: %v", err)
		return
	}
	client.HandleChannel(
		"ponginput",
		func(ctx context.Context, body io.Reader) error {
			var ret models.BrokerMessage

			decoder := json.NewDecoder(body)
			if err := decoder.Decode(&ret); err != nil {
				return err
			}

			fmt.Println(ret)

			if err := client.WriteMessage(ctx, "pongoutput", sentMsg); err != nil {
				fmt.Println(err)
				return err
			}
			return nil
		},
	)
	log.Fatal(client.Run(ctx))
}

File: clients/python/example/pingpong/pong/pong.py
from inspr import *
import sys

PONG_INPUT_CHANNEL = "ponginput"
PONG_OUTPUT_CHANNEL = "pongoutput"

def main():
    client = Client()
    msg = "Pong!"

    def readPingAndSendPong(data):
        print("data =", data, file=sys.stderr)
        print("Ping!", file=sys.stderr)
        
        try:
            client.writeMessage(PONG_OUTPUT_CHANNEL, msg)
        except:
            raise Exception


    try:
        client.writeMessage(PONG_OUTPUT_CHANNEL, msg)
    except:
        print("An error has occured")
        return

    client.handleChannel(PONG_INPUT_CHANNEL, readPingAndSendPong)
    client.run()

if __name__ == "__main__":
    main()